### PR TITLE
Refactor some common code in Pow kernel

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121973

Found when looking into https://github.com/pytorch/pytorch/issues/121926

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/cuda/PowKernel.cu [hunk 1]
```diff
@@ -168,16 +168,6 @@ void pow_tensor_scalar_kernel_impl(TensorIteratorBase& iter,
 }
 
 void pow_tensor_scalar_kernel(TensorIteratorBase& iter, const Scalar& exp_scalar) {
```
Comment (peterbell10) [inline aten/src/ATen/native/cuda/PowKernel.cu:170]: This removes the optimization from cpu-scalar exponents:
https://github.com/pytorch/pytorch/blob/014f91a9d9f94ac9a7f0711600240d7cd7f69844/aten/src/ATen/native/cuda/PowKernel.cu#L134

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: aten/src/ATen/native/cuda/PowKernel.cu (hunk 1 above)
Comment (lezcano) [reply aten/src/ATen/native/cuda/PowKernel.cu:170]: The only way to get there is through `pow_Tensor_Scalar_out` -> `pow_tensor_scalar_stub`, which is where we put all the optimisations.

Comment (peterbell10) [reply aten/src/ATen/native/cuda/PowKernel.cu:170]: The line I linked is in `pow_tensor_tensor_kernel`.

**Comment by lezcano** [comment]
To fix this I'd need to add something like
```python
  Scalar generic_scalar_value(int64_t arg) {
    _AT_DISPATCH_ALL_TYPES(common_dtype(), "scalar_value", [&] {
        return Scalar(scalar_value<scalar_t>(arg));
    });
  }
```
to TI, and then create a new dispatch for `scalar_tensor`, which would just be effectively exercised by the CUDA path. I don't think it's really worth it. Closing.

**Pull request closed without merging**